PluginCode/src/main/java/smr/cs/ualberta/libcomp/action/ReplacementAction.java

```diff
@@ -129,42 +131,71 @@ public void detectAllOpenEditors() throws IOException {
                     detectImports(psiFile, editor);
                 }
                 else {
-                    detectDependancy(editor);
+                    detectDependancy(editor, psiFile);
                 }
             }
             indexOpenEditors = indexOpenEditors + 1;
         }
     }
 
-    public void detectDependancy(@NotNull final Editor editor ) throws IOException {
+    public int detectDependenciesPSI(FileASTNode psinode)
+    {
+        int location = -1;
+        ASTNode child = psinode.getFirstChildNode();
+        String name = child.getText();
+        boolean found = name.contains("dependencies");
+        boolean over = false;
+        while ((!found) && (!over))
+        {
+            child = child.getTreeNext();
+            if (child != null)
+            {
+                name = child.getText();
+                found = name.contains("dependencies");
+            }
+            else over = true;
+           }
+        if (found)
+        {
+             location = child.getStartOffset();
+
+        }
+
+        return location;
+
+    }
+    public void detectDependancy(@NotNull final Editor editor, @NotNull final PsiFile psiFile ) throws IOException {
 
         final MarkupModel editorModel = editor.getMarkupModel();
         final Document document = editor.getDocument();
         TextAttributes attributes = EditorColorsManager.getInstance().getGlobalScheme().getAttributes(DebuggerColors.BREAKPOINT_ATTRIBUTES);
         TextAttributes softerAttributes = attributes.clone();
+        boolean dependenciesExists = false;
+        int i = 0;
+        int loc = detectDependenciesPSI(psiFile.getNode()); // Parse PSI to detect the PSI dependencies node
+        if (loc != -1) // dependencies exists
+        {
+            dependenciesExists = true;
+            i = document.getLineNumber(loc); // line number of the dependencies PSI node
+        }
+
 
         String lineText;
         String selectedTerm;
         DependListObjects.clear();
         editorModel.removeAllHighlighters();
 
-        boolean searchMode = false;
-
-        for (int i = 0; i < document.getLineCount() - 1; i++) {
+        while (dependenciesExists)
+        {
             int startOffset = document.getLineStartOffset(i);
             int endOffset = document.getLineEndOffset(i);
             lineText = document.getText(new TextRange(startOffset, endOffset)).trim();
-
-            // checking values between single quotes in dependencies in build.gradle file
-            // check te first value, "group", as the selectedTerm
-            if (searchMode) {
-                String[] valuesInQuotes = StringUtils.substringsBetween(lineText, "\'", "\'");
+            String[] valuesInQuotes = StringUtils.substringsBetween(lineText, "\'", "\'");
```

Comment:
hmm, if you have a PSI representation, why do you need to do substrings etc?

Reply:
because for gradle PSI they only detect nodes (ex: dependencies), but inside the node I still need to parse it.

Reply:
I would assume that you will get a dependencies node and then inside it can get dependency by dependency and inside each dependency get the individual attributes such as group etc ?